src/Platform/Microsoft.Testing.Platform/OutputDevice/Console/TestWorkerFrame.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+
+namespace Microsoft.Testing.Platform.OutputDevice.Console;
+
+/// <summary>
+/// Captures <see cref="TestWorker"/> that was rendered to screen, so we can only partially update the screen on next update.
+/// </summary>
+internal sealed class TestWorkerFrame
+{
+    private const int MaxColumn = 120;
+
+    private readonly (TestWorker TestWorkerProgress, int DurationLength)[] _workers;
+
+    public int Width { get; }
+
+    public int Height { get; }
+
+    public int ProgressCount { get; private set; }
+
+    public TestWorkerFrame(TestWorker?[] nodes, int width, int height)
+    {
+        Width = Math.Min(width, MaxColumn);
+        Height = height;
+
+        _workers = new (TestWorker, int)[nodes.Length];
+
+        foreach (TestWorker? status in nodes)
+        {
+            if (status is not null)
+            {
+                _workers[ProgressCount++].TestWorkerProgress = status;
+            }
+        }
+    }
+
+    public void AppendTestWorkerProgress(int i, AnsiTerminal terminal)
+    {
+        TestWorker p = _workers[i].TestWorkerProgress;
+
+        string durationString = $" ({p.Stopwatch.Elapsed.TotalSeconds:F1}s)";
+
+        _workers[i].DurationLength = durationString.Length;
+
+        int passed = p.Passed;
+        int failed = p.Failed;
+        int skipped = p.Skipped;
+
+        string? detail = !RoslynString.IsNullOrWhiteSpace(p.Detail) ? $"- {p.Detail}" : null;
+        terminal.Append('[');
+        terminal.SetColor(TerminalColor.DarkGreen);
+        terminal.Append("✅");
+        terminal.Append(passed.ToString(CultureInfo.CurrentCulture));
+        terminal.ResetColor();
+
+        terminal.Append("/");
+
+        terminal.SetColor(TerminalColor.DarkRed);
+        terminal.Append("❌");
+        terminal.Append(failed.ToString(CultureInfo.CurrentCulture));
+        terminal.ResetColor();
+
+        terminal.Append("/");
+
+        terminal.SetColor(TerminalColor.DarkYellow);
+        terminal.Append("❔");
```

Comment:
Not sure where you took the icons from, do we have something that would better represent skipped? Question mark is more of something unsure to me. Maybe something like https://glebbahmutov.com/blog/images/rt/skip1.png

Reply:
reverted the glyphs to not be emojis, maybe an arrow would work, but  the similar arrow I found has this issue filed: https://github.com/microsoft/vscode-python/issues/7705 for not being easily visible.

Reply:
![image](https://github.com/user-attachments/assets/2f6437a6-c521-488f-a4cb-c37b22a0a1e2)